injector/steamodded_injector.py

```diff
@@ -85,127 +92,145 @@ def modify_game_lua(game_lua_path):
+    ) as zip_ref:
+        zip_ref.extractall(seven_zip_dir.name)
+
+    # Check the operating system
+    # if os.name() == 'Linux':
+    #     seven_zip_path = ['wine', os.path.join(seven_zip_dir.name, "7z.exe")]
+    # elif os.name == 'nt':
+    #     seven_zip_path = os.path.join(seven_zip_dir.name, "7z.exe")
+    # else:
+    #     # Handle other operating systems or raise an error
+    #     raise NotImplementedError("This script only supports Windows and Linux.")
+
+    # Determine the operating system and prepare the 7-Zip command accordingly
+    if os.name == 'posix':
+        if platform.system() == 'Darwin':
+            # This is macOS
+            seven_zip_command = "7zz"  # Update this path as necessary for macOS
+        else:
+            # This is Linux or another POSIX-compliant OS
+            seven_zip_command = "7zz"
     else:
-        # This is Linux or another POSIX-compliant OS
-        command = "7zz"
-else:
-    # This is for Windows
-    command = f"{seven_zip_dir.name}/7z.exe"
+        # This is for Windows
+        seven_zip_command = f"{seven_zip_dir.name}/7z.exe"
+
+    return seven_zip_command, seven_zip_dir
+
+
+def main():
+    print("Starting the process...")
+
+    # URL to download the LuaJIT decompiler
+    # luajit_decompiler_url = ""
+
+    # Temporary directory for operations
+    with tempfile.TemporaryDirectory() as decompiler_dir:
+        pass
+        # This part was used to download the LuaJit decompiler
+        # luajit_decompiler_path = os.path.join(decompiler_dir, 'luajit-decompiler-v2.exe')
+
+        # # Download LuaJIT decompiler
+        # if not download_file(luajit_decompiler_url, luajit_decompiler_path):
+        #     print("Failed to download LuaJIT decompiler.")
+        #     sys.exit(1)
 
-#command = seven_zip_dir + ["x", "-o" + temp_dir.name, sfx_archive_path]
+        # print("LuaJIT Decompiler downloaded.")
 
-# seven_zip_path = os.path.join(seven_zip_dir.name, "7z.exe")
+    seven_zip_command, seven_zip_dir = download_seven_zip()
+
+    # Check if the SFX archive path is provided
+    if len(sys.argv) < 2:
+        print("Please drag and drop the SFX archive onto this executable.")
+        seven_zip_dir.cleanup()
+        sys.exit(1)
+
+    sfx_archive_path = sys.argv[1]
+    print(f"SFX Archive received: {sfx_archive_path}")
+
+    # Temporary directory for extraction and modification
+    temp_dir = tempfile.TemporaryDirectory()
+    print(temp_dir.name)
+    # Extract the SFX archive
+    subprocess.run([seven_zip_command, "x", f"-o{temp_dir.name}", sfx_archive_path], check=True)
+    print("Extraction complete.")
+
+    # Determine the base directory (where the .exe is located)
+    if getattr(sys, "frozen", False):
+        # Running in a PyInstaller or Nuitka bundle
+        base_dir = os.path.dirname(sys.executable)
+    else:
+        # Running in a normal Python environment
+        base_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..")
```

Comment:
`base_dir = os.path.dirname(os.path.abspath(__file__))`

To match new file structure